pkg/dashboard/resource/function.go

```diff
@@ -51,7 +51,7 @@ type functionInfo struct {
 }
 
 func (fr *functionResource) ExtendMiddlewares() error {
-	fr.resource.addAuthMiddleware()
+	fr.resource.addAuthMiddleware(auth.Options{})
```

Comment:
pass a reference or na interface, not a struct directly.

a side tip- a nicer way to do it by the way is have `addAuthMiddleware` accepts multiple options whereas each single option is a function that returns an interface. that interface has an `Apply` that mutates whatever it needs before performing the action. In this case it requires a bigger refactor which isnt suggested here.